Source/Connector.swift

```diff
@@ -90,17 +90,13 @@ class Connector {
                 return Disposables.create()
             }
 
-            guard !peripheral.isConnected else {
-                observer.onError(BluetoothError.peripheralAlreadyConnected(peripheral))
-                return Disposables.create()
-            }
-
-            let connectingStarted = strongSelf.connectingBox.compareAndSet(
-                compare: { !$0.contains(peripheral.identifier) },
-                set: { $0.append(peripheral.identifier) }
+            let connectionBlocked = strongSelf.connectingBox.compareAndSet(
+                compare: { !peripheral.isConnected && !$0.contains(peripheral.identifier) },
+                set: { $0.insert(peripheral.identifier) }
             )
-            guard connectingStarted else {
-                observer.onError(BluetoothError.peripheralIsConnecting(peripheral))
+            
+            guard connectionBlocked else {
```

Comment:
what you think about changing name `connectionBlocked` to something like `connectingStarted`. For me it would be more readable